.engine_scripts/playwright/onReady.js

```diff
@@ -1,35 +1,22 @@
-const fs = require('fs');
 const autoScroll = require('../auto-scroll');
-const scrollTop = require('../scroll-top');
 
 module.exports = async (page, scenario, viewport, isReference, browserContext) => {
-  console.log('SCENARIO > ' + scenario.label);
-
-  if (scenario.useCssOverride) {
-    await require('./overrideCSS')(page, scenario);
-  }
-
-  const jsOnReadyPath = scenario.jsOnReadyPath;
+  await require('./embedFiles')(scenario, page);
+  await page.evaluate(autoScroll);
 
-  if (!jsOnReadyPath) {
-    return;
-  } else if (!fs.existsSync(jsOnReadyPath)) {
-    console.log('File not exist: ' + jsOnReadyPath);
-  } else {
-    const jsOnReadyScript = fs.readFileSync(jsOnReadyPath, 'utf-8');
-    await page.evaluate(jsOnReadyScript).then(() => console.log('onReady script executed for: ' + scenario.label));
-  }
+  page.on('load', async (data) => {
+    await require('./embedFiles')(scenario, data);
+    await data.evaluate(require('../auto-scroll'));
+  });
 
-  await page.evaluate(autoScroll);
+  console.log('SCENARIO > ' + scenario.label);
 
   if (!!scenario.actions) {
     await require('./actions')(page, scenario);
   } else {
     await require('./clickAndHoverHelper')(page, scenario);
   }
 
-  await page.evaluate(scrollTop);
```

Comment:
This line is used for scroll to Top after all user interaction is finished.
For example when user fill a form at the bottom of the page, our tool will scroll the form into view. If we don't scroll to the top again, the tool will capture the screenshot at current position, this will cause some side effect such as the fixed header being capture at the middle of the page instead of in the top.